.ddev/config.yaml

```diff
@@ -20,14 +19,17 @@ web_extra_exposed_ports:
 # https://github.com/vitejs/vite/discussions/3396
 web_extra_daemons:
   - name: astro-dev-daemon
-    command: "sleep 3 && npm run dev -- --host" # sleep avoids race conditions
+    command: bash -c 'npm install && touch /var/tmp/npminstalldone && npm run dev -- --host'
     directory: /var/www/html
 hooks:
   post-start:
-    - exec-host: echo -e "                                  NOTICE                                        \n
+    - exec: bash -c 'while [ ! -f /var/tmp/npminstalldone ]; do sleep 1; done'
+    - exec: npm run build
```

Comment:
Why isn't this in the daemon startup instead of here? Another `&&` ?

Reply:
great question, I was going to add it there but npm run build takes 30seconds.
<img width="702" alt="Screenshot 2024-03-14 at 2 24 05 PM" src="https://github.com/ddev/ddev.com/assets/39039024/c8fc4792-f47f-43e4-96c9-5b834862ff29">
and since it on that background daemon it creates a lag plus we only need it the first time around since most times we be running HMR(hot module reloading anyway).

How do I turn the echo -e green? I wanted ${DDEV_PRIMARY_URL}:4321 that to be green to but i can't find the variable.

Reply:
Pretty fancy, try https://stackoverflow.com/questions/5947742/how-to-change-the-output-color-of-echo-in-linux

Reply:
`echo -e "\e[1;31mHello \e[1;35mWorld"`
that works locally but once it goes through ddev exec it fails.
ddev exec echo -e '\e[1;31mHello \e[1;35mWorld'
e[1
bash: line 1: 31mHello: command not found
Failed to execute command echo -e \e[1;31mHello \e[1;35mWorld: exit status 127

Reply:
Try it in single quotes.

Reply:
<img width="690" alt="Screenshot 2024-03-14 at 4 13 19 PM" src="https://github.com/ddev/ddev.com/assets/39039024/e6ff55bd-4b89-4fe1-880f-9db696f83136">
Got it. I must have been having a typo there.